lib/sass/importers/filesystem.rb

```diff
@@ -154,7 +154,9 @@ def find_real_file(dir, name, options)
             [Sass::Util.cleanpath(full_path).to_s, s]
           end
         end.flatten(1)
-        return if found.empty?
+        if found.empty? && _extension?(name) && File.directory?("#{dir}/#{name}")
```

Comment:
I don't understand what `_extension?(name)` is doing here... are we only checking for an index file if `name` has an extension? That seems wrong.

Reply:
We figured it would be more readable than `split(name)[2].nil?` I'll switch back

Reply:
I meant more "why do we need to check for an extension at all" than "why is it factored out into a private method"

Reply:
Because of your above comment that if it has an extension in the name we shouldn't look at all.

Reply:
I just meant a `.sass` or `.scss` extension. If we disallow *any* extension, [this spec](https://github.com/john681611/sass-spec/tree/bd1e7759b7641cc88b4f1a49b7058e08e5a48b42/spec/directives/import/index/dir_dot_foo) will fail (and in fact I'm not sure why it isn't failing already...)

Reply:
Oh, sorry, I was being dumb here... I didn't realize that this was using the `split()` method defined below that already checks for the correct file extensions. Your original code was fine; I'll revert b196d9c.